File: tools/ADLINK/GIE Series/Sample/PoE Samples/C#/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace SmartPoESample
{
    public partial class Form1 : Form
    {
        short m_Number = -1;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ushort i;
            short num;
            for (i = 0; i < SmartPoE.MAX_CARD; i++)
            {
                if ((num = SmartPoE.SmartPoE_Register_Card(i)) != 0)
                    break;

                SmartPoE.SmartPoE_Release_Card((ushort)num);

                cboCardNumber.Items.Add(i.ToString());
            }

            if (i == 0)
            {
                MessageBox.Show("No card is found!");
                Close();
                return;
            }

            cboCardNumber.SelectedIndex = 0;
        }

        private void Form1_FormClosed(object sender, FormClosedEventArgs e)
        {
            if (m_Number >= 0)
                SmartPoE.SmartPoE_Release_Card((ushort)m_Number);
        }

        private void cboCardNumber_SelectedIndexChanged(object sender, EventArgs e)
        {
            short Number;
            ushort card_num = (ushort)cboCardNumber.SelectedIndex;

            if(m_Number >= 0)
                SmartPoE.SmartPoE_Release_Card((ushort)m_Number);

            Number = SmartPoE.SmartPoE_Register_Card(card_num);
            if (Number != 0)
            {
                MessageBox.Show("Selecting card is failed");
                return;
            }

            ushort ID = 0;
            short ret = SmartPoE.SmartPoE_Get_ID((ushort)Number, out ID);
            if (ret < 0)
                txtCardID.Text = "Error";
            else
                txtCardID.Text = ID.ToString();
            
            m_Number = (short)Number;
        }

        private void btnSet_Click(object sender, EventArgs e)
        {
            if (m_Number < 0)
                return;

            if (!CheckUserInput())
            {
                MessageBox.Show("Some ON/OFF state of ports are not set!");
                return;
            }

            ushort EnPort1 = (ushort)(rdoOn1.Checked ? 1 : 0);
            ushort EnPort2 = (ushort)(rdoOn2.Checked ? 1 : 0);
            ushort EnPort3 = (ushort)(rdoOn3.Checked ? 1 : 0);
            ushort EnPort4 = (ushort)(rdoOn4.Checked ? 1 : 0);

            if (SmartPoE.SmartPoE_Power_Enable((ushort)m_Number, EnPort1, EnPort2, EnPort3, EnPort4) != 0)
                MessageBox.Show("Setting power is failed");
        }

        private bool CheckUserInput()
        {
            ushort EnPort1On = (ushort)(rdoOn1.Checked ? 1 : 0);
            ushort EnPort2On = (ushort)(rdoOn2.Checked ? 1 : 0);
            ushort EnPort3On = (ushort)(rdoOn3.Checked ? 1 : 0);
            ushort EnPort4On = (ushort)(rdoOn4.Checked ? 1 : 0);

            ushort EnPort1Off = (ushort)(rdoOff1.Checked ? 1 : 0);
            ushort EnPort2Off = (ushort)(rdoOff2.Checked ? 1 : 0);
            ushort EnPort3Off = (ushort)(rdoOff3.Checked ? 1 : 0);
            ushort EnPort4Off = (ushort)(rdoOff4.Checked ? 1 : 0);

            if ((EnPort1On == 0 && EnPort1Off == 0) ||
                (EnPort2On == 0 && EnPort2Off == 0) ||
                (EnPort3On == 0 && EnPort3Off == 0) ||
                (EnPort4On == 0 && EnPort4Off == 0))
                return false;

            return true;

        }
    }
}

File: tools/ADLINK/GIE Series/Sample/PoE Samples/VB.Net/Form1.vb
Imports System.Math

Public Class Form1

    Private m_Number As Short = -1

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        Dim i As UShort
        Dim num As Short

        For i = 0 To SmartPoE.MAX_CARD - 1

            num = SmartPoE.Register_Card(i)
            If num <> 0 Then
                Exit For
            End If

            SmartPoE.Release_Card(num)

            cboCardNumber.Items.Add(i.ToString())

        Next

        If i = 0 Then

            MessageBox.Show("No card is found!")
            Close()
            Exit Sub

        End If

        cboCardNumber.SelectedIndex = 0

    End Sub

    Private Sub Form1_FormClosed(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosedEventArgs) Handles MyBase.FormClosed

        If m_Number >= 0 Then
            SmartPoE.Release_Card(m_Number)
        End If

    End Sub

    Private Sub cboCardNumber_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cboCardNumber.SelectedIndexChanged
        Dim Number As Short
        Dim card_num As UShort = cboCardNumber.SelectedIndex

        If m_Number >= 0 Then
            SmartPoE.Release_Card(m_Number)
        End If

        Number = SmartPoE.Register_Card(card_num)
        If Number <> 0 Then

            MessageBox.Show("Selecting card is failed")
            Exit Sub
        End If

        Dim ID As UShort = 0
        Dim ret As Short = SmartPoE.Get_ID(Number, ID)

        If ret < 0 Then
            txtCardID.Text = "Error"
        Else
            txtCardID.Text = ID.ToString()
        End If

        m_Number = Number

    End Sub

    Private Sub btnSet_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSet.Click

        If m_Number < 0 Then
            Exit Sub
        End If

        If Not CheckUserInput() Then
            MessageBox.Show("Some On/OFF state of ports are not set!")
            Exit Sub
        End If

        Dim EnPort1 As UShort = Abs(CInt(rdoOn1.Checked))
        Dim EnPort2 As UShort = Abs(CInt(rdoOn2.Checked))
        Dim EnPort3 As UShort = Abs(CInt(rdoOn3.Checked))
        Dim EnPort4 As UShort = Abs(CInt(rdoOn4.Checked))

        If SmartPoE.Power_Enable(m_Number, EnPort1, EnPort2, EnPort3, EnPort4) <> 0 Then
            MessageBox.Show("Setting power is failed")
        End If

    End Sub

    Private Function CheckUserInput() As Boolean

        Dim EnPort1On As UShort = Abs(CInt(rdoOn1.Checked))
        Dim EnPort2On As UShort = Abs(CInt(rdoOn2.Checked))
        Dim EnPort3On As UShort = Abs(CInt(rdoOn3.Checked))
        Dim EnPort4On As UShort = Abs(CInt(rdoOn4.Checked))

        Dim EnPort1Off As UShort = Abs(CInt(rdoOff1.Checked))
        Dim EnPort2Off As UShort = Abs(CInt(rdoOff2.Checked))
        Dim EnPort3Off As UShort = Abs(CInt(rdoOff3.Checked))
        Dim EnPort4Off As UShort = Abs(CInt(rdoOff4.Checked))

        If (EnPort1On = 0 And EnPort1Off = 0) Or _
            (EnPort2On = 0 And EnPort2Off = 0) Or _
            (EnPort3On = 0 And EnPort3Off = 0) Or _
            (EnPort4On = 0 And EnPort4Off = 0) Then
            Return False
        End If

        Return True

    End Function
End Class
